code-generators/src/main/java/net/minestom/codegen/CodeGenerator.java

```diff
@@ -75,8 +75,7 @@ public void generateKeys(InputStream resourceFile, String packageName, String ty
         }
 
         ClassName typeClass = ClassName.bestGuess(packageName + "." + typeName); // Use bestGuess to handle nested class
-        ClassName registryKeyClass = ClassName.get("net.minestom.server.registry", "DynamicRegistry", "Key");
-        ParameterizedTypeName typedRegistryKeyClass = ParameterizedTypeName.get(registryKeyClass, typeClass);
+        ClassName registryKeyClass = ClassName.get("net.kyori.adventure.key", "Key");
```

Comment:
In the future, we should add `Adventure `as a general dependency for the generator. This would allow us to remove the `ClassName.get` call for the Key class. While the current code is functional, relying on strings can lead to issues if the associated class from `Adventure` is renamed